File: frontend/src/api/shop.ts
import request from "./request";

export interface Shop {
  id: number;
  name: string;
  category: string;
  address: string;
  region: string;
  rating: number;
  reviewCount: number;
  avgPrice: number;
  imageUrl: string;
  description: string;
  status: number;
  createdAt: string;
  updatedAt: string;
}

export const getShopList = () => {
  return request.get("/shop/list");
};

export const getShopDetail = (id: number | string) => {
  return request.get(`/shop/${id}`);
};

File: backend/src/main/java/com/mlog/entity/Shop.java
package com.mlog.entity;

import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
public class Shop {

    private Long id;

    private String name;

    private String category;

    private String address;

    private String region;

    private String imageUrl;

    private BigDecimal rating;

    private Integer reviewCount;

    private Integer avgPrice;

    private String description;

    private Integer status;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    private Double latitude;

    private Double longitude;

    private Double distance;
}
